actors/miner/src/types.rs

```diff
@@ -145,6 +148,53 @@ pub struct ExpirationExtension {
     pub sectors: UnvalidatedBitField,
     pub new_expiration: ChainEpoch,
 }
+#[allow(clippy::too_many_arguments)] // validate mut prevents implementing From
+impl Into<ValidatedExpirationExtension> for ExpirationExtension2 {
+    fn into(mut self) -> ValidatedExpirationExtension {
+        let mut sectors = BitField::new();
+        for sc in self.sectors_with_claims {
+            sectors.set(sc.sector_number)
+        }
+        sectors |= self.sectors.validate_mut().unwrap();
+
+        ValidatedExpirationExtension {
+            deadline: self.deadline,
+            partition: self.partition,
+            sectors,
+            new_expiration: self.new_expiration,
+        }
+    }
+}
+
+#[derive(Serialize_tuple, Deserialize_tuple)]
+pub struct ExtendSectorExpiration2Params {
+    pub extensions: Vec<ExpirationExtension2>,
+}
+
+impl Cbor for ExtendSectorExpiration2Params {}
+
+#[derive(Serialize_tuple, Deserialize_tuple, Clone)]
+pub struct SectorClaim {
+    pub sector_number: SectorNumber,
+    pub maintain_claims: Vec<ClaimID>,
+}
+
+#[derive(Serialize_tuple, Deserialize_tuple)]
+pub struct ExpirationExtension2 {
+    pub deadline: u64,
+    pub partition: u64,
+    pub sectors: UnvalidatedBitField, // IDs of sectors without FIL+ claims
+    pub sectors_with_claims: Vec<SectorClaim>,
+    pub new_expiration: ChainEpoch,
+}
+
+#[derive(Clone)]
+pub struct ValidatedExpirationExtension {
+    pub deadline: u64,
+    pub partition: u64,
+    pub sectors: BitField,
+    pub new_expiration: ChainEpoch,
+}
```

Comment:
I think the intention with these files has been to define the exported API, of which this is not a part. Move it into the `lib.rs`, or to another peer file if there's an opportunity to move the related functions that use this type at the same time.